### DISCUSSION TITLE
Laser engraving on topographic map

### DISCUSSION THREAD

**Discussion by SnowBeaver** [martin2250/OpenCNCPilot#175] [General]
Greetings,
Looking for a solution to engrave information on a topographic map. I was lead here from a discussion in lasergrbl. [https://github.com/arkypita/LaserGRBL/issues/624#issuecomment-487268633](url). Height difference across the piece is 10mm so a mean to compensate for height is required to maintain the laser beam focused. 
The height map used to cnc the piece comes from a grayscale bitmap image, where the gray level represent height. File size for my actual project is 800k (attached), but could be made smaller as there is no need for pixel sizes precision to maintain proper focal height. 
1- Can bitmap file be use directly as height map in opencncpilot?
2- Is it possible to drive a laser instead of router?

Cheers
[lac-kenogami-2.zip](https://github.com/martin2250/OpenCNCPilot/files/11021828/lac-kenogami-2.zip)

**Comment by martin2250** [comment]
Hi @SnowBeaver,

1: no, you'd need to convert the image to a height map file first. It is a simple xml-based format, so if you know any programming language, it should not take very long. If you want to look at the file format, select the "create height map from mathematical formula" option when creating a new height map in OpenCNCPilot and save it afterwards.

2: OpenCNCPilot just passes the GCode to grbl, so it should work fine. Maybe lasers use some additional unsupported GCodes, but those should just be left unmodified.

Cheers,
Martin
